### ISSUE TITLE
[Required] onunload leaf detach antipattern

### ISSUE THREAD

**Issue by mssoftjp** [mssoftjp/obsidian-ai-transcriber#21]
## Issue概要
ObsidianReviewBotから指摘されたRequired項目。onunloadメソッドでleaf.detach()を呼び出すアンチパターンを回避する。

## 指摘内容
> **Required:** [[1](https://github.com/mssoftjp/obsidian-ai-transcriber/blob/01b4ce4d057316450bd84294a104a63ac5196e15/src/main-api.ts#L36)] Avoid calling leaf.detach() in onunload, it's done automatically by Obsidian.

## 対象ファイル
- `src/main-api.ts` (Line 36)

## 問題詳細
onunloadメソッド内でleaf.detach()を手動で呼び出すと、Obsidianの自動クリーンアップと競合し、予期しない動作を引き起こす可能性がある。

## 実装計画
1. main-api.tsのonunloadメソッドを確認
2. 不適切なleaf.detach()呼び出しを削除
3. 適切なリソースクリーンアップのみ実装
4. プラグインライフサイクルの最適化

## 修正例
```typescript
// Before (アンチパターン)
onunload() {
  this.leaves.forEach(leaf => leaf.detach()); // ❌ 避けるべき
  this.cleanup();
}

// After (適切な実装)
onunload() {
  // Obsidianが自動的にleafをdetachするため、手動detachは不要
  this.cleanup();
}
```

## 対応状況
✅ **COMPLETED**

実装確認：
- main-api.tsのonunloadメソッドが適切に実装されている
- leaf.detach()の不適切な呼び出しは含まれていない
- ResourceManagerとStatusBarManagerの適切なクリーンアップを実装

## チェック項目
- [x] onunloadメソッドでのleaf.detach()呼び出し確認
- [x] 適切なリソースクリーンアップの実装
- [x] プラグインライフサイクルの検証
- [x] Obsidianのベストプラクティス準拠

参考: https://github.com/obsidianmd/obsidian-releases/pull/7229#issuecomment-3132987462
